sotodlib/toast/ops/sim_source.py

```diff
@@ -0,0 +1,694 @@
+import os
+import pickle
+import copy
+
+import traitlets
+
+import numpy as np
+
+from astropy import constants
+from astropy import units as u
+
+import ephem
+
+from scipy.interpolate import RectBivariateSpline
+
+from toast.timing import function_timer
+
+from toast import qarray as qa
+
+from toast.data import Data
+
+from toast.traits import trait_docs, Int, Unicode, Float, Instance, List, Bool, Quantity
+
+from toast.ops.operator import Operator
+from toast.instrument import Focalplane
+
+from toast.utils import Logger, Timer
+
+from toast.observation import default_values as defaults
+
+from ...coords.local import *
+
+
+
+def tb2s(tb, nu):
```

Comment:
This function looks the same as the one defined in `toast.ops.sim_sso.py`.  We should consolidate that to one function and just import it from the other source file.  In toast3, we support `astropy.units` everywhere (in toast2 we were very loose about that).  I think we can modify these functions (in `sim_sso.py`) to propagate the units until we have to convert to a value.  This would address the point that @mhasself was getting at.

Reply:
I moved the functions to a utils file and kept the units. There is no check on the output because the input units are set so that the output should be the expected one